media/server/gstplayer/source/GstGenericPlayer.cpp

```diff
@@ -868,14 +868,14 @@ void GstGenericPlayer::pushSampleIfRequired(GstElement *source, const std::strin
         // Sending initial sample not needed
         return;
     }
-    for (const auto &[position, resetTime, appliedRate] : initialPosition->second)
+    for (const auto &[position, resetTime, appliedRate, stopPosition] : initialPosition->second)
     {
         GstSeekFlags seekFlag = resetTime ? GST_SEEK_FLAG_FLUSH : GST_SEEK_FLAG_NONE;
         RIALTO_SERVER_LOG_DEBUG("Pushing new %s sample...", typeStr.c_str());
         GstSegment *segment{m_gstWrapper->gstSegmentNew()};
         m_gstWrapper->gstSegmentInit(segment, GST_FORMAT_TIME);
         if (!m_gstWrapper->gstSegmentDoSeek(segment, m_context.playbackRate, GST_FORMAT_TIME, seekFlag,
-                                            GST_SEEK_TYPE_SET, position, GST_SEEK_TYPE_SET, GST_CLOCK_TIME_NONE, nullptr))
+                                            GST_SEEK_TYPE_SET, position, GST_SEEK_TYPE_SET, stopPosition, nullptr))
```

Comment:
Should there be a code comment to say that we expect
kUndefinedPosition  == GST_CLOCK_TIME_NONE

Reply:
But (if you want) I can add a testcase that EXPECT_EQ(kUndefinedPosition, GST_CLOCK_TIME_NONE);

Reply:
test added